### PR TITLE
[topgen] Fix whitespace errors for crossbar generation

### PR DESCRIPTION
**Pull request by a-will**

### CODE REVIEW

File: util/tlgen/xbar_cover.cfg.tpl
Comment (tjaychen) [inline util/tlgen/xbar_cover.cfg.tpl:52]: hmm... the `fix_trailing_whitespace.py` is supposed to ignore template files, are you sure you needed this one?

File: hw/top_earlgrey/ip/xbar_peri/dv/autogen/xbar_cover.cfg
Comment (tjaychen) [inline hw/top_earlgrey/ip/xbar_peri/dv/autogen/xbar_cover.cfg:130]: starting to think we should ignore cfg files too heh.

**Review by tjaychen — APPROVED** [review]

File: util/tlgen/xbar_cover.cfg.tpl
Comment (a-will) [reply util/tlgen/xbar_cover.cfg.tpl:52]: It's to fix the generated files, which it _does_ check (when they change).

Comment (tjaychen) [reply util/tlgen/xbar_cover.cfg.tpl:52]: o that's right. Should we exclude `.gen.` also? :)

Comment (a-will) [reply util/tlgen/xbar_cover.cfg.tpl:52]: If I am reading it right, we currently only exclude vendor/ directories. Everything else is checked. Hm... I don't have strong opinions about which things need to be exempted, though--I just didn't like that it was a new check with a delayed trigger, and it exploded in my face / PR, haha.

**Comment by rswarbrick** [comment]
@a-will: For next time, please could you avoid "Autogenerate" commits like the last one in this PR? Once the PR is merged, it gets flattened and we've now got an inconsistent commit in the master branch history.

This is a little awkward, but I think auto-generating as part of a script update is the best solution given the constraint that we're checking in auto-generated files.

**Comment by a-will** [comment]
> @a-will: For next time, please could you avoid "Autogenerate" commits like the last one in this PR? Once the PR is merged, it gets flattened and we've now got an inconsistent commit in the master branch history.
> 
> This is a little awkward, but I think auto-generating as part of a script update is the best solution given the constraint that we're checking in auto-generated files.

You bet! I noticed someone else do it this way and thought that might be the group's norm. Apparently not, haha 😂

There might be ways in the future for us to avoid checking in autogenerated files (which is the approach I am more familiar with), but while we're checking them in, doing the generation in the same commit makes sense to me.

**Comment by rswarbrick** [comment]
Thanks!

**Comment by msfschaffner** [comment]
@a-will I think what we've done sometimes is to split the autogenerated part into a separate commit for review purposes only (to reduce clutter) and then squash that commit into the script updates once the PR is approved. There is however still a chance that the author forgets the squashing step before merging, so I would recommend to split the changes only for really large autogen diffs and otherwise put the changes into the same commit right away.

**Pull request merged**
